### DISCUSSION TITLE
How do you use sniprun ? (as the author) I'm quite curious

### DISCUSSION THREAD

**Discussion by michaelb** [michaelb/sniprun#275] [Show and tell]
I've had fellas in the issues sections talking about running Ansible playbook, pushing the API to its limits, running python ML exercises and I myself used it for a interview coding test (though I should rather have been copying from stackoverflow).

What are the fun/interesting/stupid ways you've used Sniprun ?

**Comment by RyanParsley** [comment]
I like the idea of a markdown page that links up an image and has a code snippet in the file that is responsible for generating the asset. For instance, shell code that runs gnuplot against a data set that may update. That Gnuplot snippet renders an SVG. The tricky bit is grappling with your present working directory with respect to the script and any assets you're trying to generate. 

It feels very powerful when it works and hard to evangelize when I'm debugging it :D

I'm just learning about literate code and SnipRun feels like a great tool to enable this in more clever documentation... but I struggle to work the kinks out of the workflow.

**Reply by michaelb** [reply]
ahah you're definitively off the beaten track
SnipRun is basically intended to give you back a 'text' result within Neovim. That being said, I don't see a fundamental reason why this wouldn't work.
Don't hesitate to open an github issue if you face any problem, and for debugging, ~/.cache/sniprun/* is your friend
